discovery-infra/tests/test_logs.py

```diff
@@ -0,0 +1,104 @@
+        self.verify_no_logs_uploaded(cluster_id, api_client, path)
+        
+        #install the cluster
+        started_cluster_install_at = time.time()
+        self.install_cluster_and_wait(cluster_id, api_client)
+
+        #download logs into a file. At this point we are expecting to have logs from each
+        #host plus controller logs
+        logs_collected_at = utils.get_logs_collected_at(api_client, cluster_id)
+        expected_min_log_num = len(node_controller.list_nodes()) + 1
+        self.verify_logs_uploaded(cluster_id, api_client, path, started_cluster_install_at, expected_min_log_num, logs_collected_at)
+
+    
+    def test_collect_logs_on_failure(self, api_client, node_controller, cluster):
+        '''cancel insllation after at least one host is booted and check that logs are uploaded'''
+        # Define new cluster and prepare it for installation
+        cluster_id = cluster().id
+        self.setup_hosts(cluster_id, api_client, node_controller)
+        self.prepare_for_installation(cluster_id, api_client, node_controller)
+
+        started_cluster_install_at = time.time()
+        self.start_cluster_install(cluster_id=cluster_id, api_client=api_client)
+        # Cancel cluster install once at least one host has been rebooted
+        self.wait_for_one_host_to_boot_during_install(cluster_id=cluster_id, api_client=api_client)
+        self.cancel_cluster_install(cluster_id=cluster_id, api_client=api_client)
+        assert self.is_cluster_in_cancelled_status(
+            cluster_id=cluster_id,
+            api_client=api_client
+        )
+         #download logs into a file. At this point logs exist and we expect them to
+         #be from the hosts that finished reboot without controller logs
+        path = "/tmp/test_on_cancel_logs.tar"
+        expected_min_log_num = 1
+        logs_collected_at = utils.get_logs_collected_at(api_client, cluster_id)
+        self.verify_logs_uploaded(cluster_id, api_client, path, started_cluster_install_at, expected_min_log_num, logs_collected_at)
+
+    def verify_no_logs_uploaded(self, cluster_id, api_client, path):
+        with pytest.raises(ApiException) as ex:
+            api_client.download_cluster_logs(cluster_id, path)
+        assert "No log files" in str(ex.value)
+
+    def verify_logs_uploaded(self, cluster_id, api_client, path, started_cluster_install_at, expected_min_log_num, logs_collected_at):
+        #download logs
+        api_client.download_cluster_logs(cluster_id, path)
+        #verify that logs were collected from all expected sources
+        assert(os.path.exists(path))
+        try:
+            tar = tarfile.open(path)
+            logging.info(f'downloaded logs: {tar.getnames()}')
+            assert len(tar.getnames()) >= expected_min_log_num
+            tar.close()
+            #verify that logs are recent
+            for collected_at in logs_collected_at:
+                 if collected_at > time.time() - 86400000:
```

Comment:
not sure i understand what is this int?

Reply:
i think u need to verify that collected_at is bigger than starting_installation time, just to be sure those are latest logs and relevant to specific installation

Reply:
In the intermdeidate state not all host has valid collected_at so this filters the illegal timestamps out

Reply:
as i wrote in slack, the next one that will create new test that state will be different in it. for example get logs after first installation was reset. This func will not validate it right in case of bug. So why just not to take time before installation in UTC and validate on it?